cmd/template/cluster/provider/common.go

```diff
@@ -51,3 +64,80 @@ func newCAPIV1Alpha3ClusterCR(config ClusterCRsConfig, infrastructureRef *corev1
 
 	return cluster
 }
+
+func runMutation(ctx context.Context, client k8sclient.Interface, input []byte) ([]byte, error) {
+	obj := &unstructured.Unstructured{}
+	dec := apiyaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	_, gvk, err := dec.Decode(input, nil, obj)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Create a DiscoveryClient to get the GVR.
+	dc, err := discovery.NewDiscoveryClientForConfig(client.RESTConfig())
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Mapping needed for the dynamic client.
+	mapping, err := findGVR(gvk, dc)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Create a DynamicClient to execute our request.
+	dyn, err := dynamic.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return []byte{}, err
+	}
+
+	namespace, namespaced, err := unstructured.NestedString(obj.Object, "metadata", "namespace")
```

Comment:
This works to figure out if a resource is namespaced - we just have to be sure to do it right in the template.

Reply:
This feels pretty fragile. If it's not too much trouble, I'd prefer to get the resource definition from the API in the same way as `kubectl api-resources`. I'm not sure how to do this in code, but for example `GET:/apis/release.giantswarm.io/v1alpha1` returns `{"kind":"APIResourceList","apiVersion":"v1","groupVersion":"release.giantswarm.io/v1alpha1","resources":[{"name":"releases","singularName":"release","namespaced":false,"kind":"Release","verbs":["delete","deletecollection","get","list","patch","create","update","watch"],"categories":["common","giantswarm"],"storageVersionHash":"W2y3iU4kvpQ="},{"name":"releases/status","singularName":"","namespaced":false,"kind":"Release","verbs":["get","patch","update"]}]}` which says `"namespaced":false`. What do you think?